pkg/app/piped/cloudprovider/cloudrun/client.go

```diff
@@ -62,27 +62,103 @@ func newClient(ctx context.Context, projectID, region, credentialsFile string, l
 	return c, nil
 }
 
-func (c *client) Apply(ctx context.Context, sm ServiceManifest) (*Service, error) {
-	service, err := manifestToRunService(sm)
+func (c *client) Create(ctx context.Context, sm ServiceManifest) (*Service, error) {
+	svcCfg, err := manifestToRunService(sm)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		svc    = run.NewNamespacesServicesService(c.client)
+		parent = makeCloudRunParent(c.projectID)
+		call   = svc.Create(parent, svcCfg)
+	)
+	call.Context(ctx)
+
+	service, err := call.Do()
+	if err != nil {
+		if e, ok := err.(*googleapi.Error); ok && e.Code == http.StatusNotFound {
+			return nil, fmt.Errorf("failed to create service: code=%d, message=%s, details=%s", e.Code, e.Message, e.Details)
```

Comment:
nits, not sure can we get `http.StatusNotFound` here since we're going to create this service 🤔

Reply:
Nice catch! Copy/pasted from the update. lol

Reply:
Applied!